vars/checkRequestAssignReleaseOwners.groovy

```diff
@@ -0,0 +1,199 @@
+ */
+void call(Map args = [:]) {
+    def inputManifest = args.inputManifest
+    String action = args.action ?: 'check'
+    def version = extractVersionFromManifest(inputManifest[0])
+    def now = LocalDate.now()
+    def monthYear = String.format("%02d-%d", now.monthValue, now.year)
+    def maintainersIndex = "maintainer-inactivity-${monthYear}"
+
+    // Parameter validation
+    validateParameters(args)
+
+    List<String> componentsMissingReleaseOwners = []
+
+    inputManifest.each { inputManifestFile ->
+        def inputManifestObj = readYaml(file: inputManifestFile)
+        withCredentials([
+                string(credentialsId: 'jenkins-health-metrics-account-number', variable: 'METRICS_HOST_ACCOUNT'),
+                string(credentialsId: 'jenkins-health-metrics-cluster-endpoint', variable: 'METRICS_HOST_URL')]) {
+            withAWS(role: 'OpenSearchJenkinsAccessRole', roleAccount: "${METRICS_HOST_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-session') {
+                def metricsUrl = env.METRICS_HOST_URL
+                def awsAccessKey = env.AWS_ACCESS_KEY_ID
+                def awsSecretKey = env.AWS_SECRET_ACCESS_KEY
+                def awsSessionToken = env.AWS_SESSION_TOKEN
+
+                ReleaseMetricsData releaseMetricsData = new ReleaseMetricsData(metricsUrl, awsAccessKey, awsSecretKey, awsSessionToken, version, this)
+                ComponentRepoData componentRepoData = new ComponentRepoData(metricsUrl, awsAccessKey, awsSecretKey, awsSessionToken, version, maintainersIndex, this)
+
+                inputManifestObj.components.each { component ->
+                    def releaseOwner = releaseMetricsData.getReleaseOwners(component.name)
+                    if (releaseOwner == null || releaseOwner.isEmpty()) {
+                        componentsMissingReleaseOwners.add(component.name)
+                        if (action != 'check') {
+                            handleMissingReleaseOwner(component, releaseMetricsData, componentRepoData, action)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if (!componentsMissingReleaseOwners.isEmpty()) {
+        echo("Components missing release owner: ${componentsMissingReleaseOwners}")
+    } else {
+        echo('All components have release owner assigned.')
+    }
+}
+
+/**
+ * Validates input parameters
+ */
+private void validateParameters(Map args) {
+    if (!args.inputManifest || args.inputManifest.isEmpty()) {
+        error "inputManifest parameter is required."
+    } else {
+        args.inputManifest.each { inputManifestFile ->
+            if (!fileExists(inputManifestFile)) {
+                error("Invalid path. Input manifest file does not exist at ${inputManifestFile}")
+            }
+        }
+    }
+
+    String action = args.action
+    println("Action is: ${action}")
+    List<String> validActions = ['check', 'assign', 'request']
```

Comment:
Where are we implementing check action here Sayali? 
Is it check and assign or check and request based on release time line?

Reply:
Just will just output the the components missing the release owners on the console. What we want to do with it, we can iterate later.